### ISSUE TITLE
Socket connection timeout

### ISSUE THREAD

**Issue by missinglink** [pelias/microservice-wrapper#36]
It would be great if the requests would timeout instantly in the case where the socket connection was not possible, such as when the port is closed or the host is down.

I had a quick look at the docs for `superagent`, and while they have two different `timeout` settings, neither of them seems to cover this case.

It would allow us to timeout much faster when a service was configured but unavailable due to misconfiguration or outage.

related: https://github.com/pelias/kubernetes/pull/81

**Comment by orangejulius** [comment]
Yeah this is interesting, this might be something that is only a problem for us with K8S. Worst case, we might want to warn people that if they are not going to use all pelias services, they should ensure that the unused services are disabled in `pelias.json`, for example by using [enabled: false](https://github.com/pelias/kubernetes/blob/master/values.yaml#L50) in the Helm config.
